### PR TITLE
Olenka mysko

### PR DESCRIPTION
**Pull request by kolyasalubov**

### CODE REVIEW

File: HW1/Olenka_M/HomeWork1.1.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+print("\n--------------task 1.1---------------\n")
+
```
Comment (kolyasalubov) [inline HW1/Olenka_M/HomeWork1.1.py:2]: good job

File: HW1/Olenka_M/HomeWork1.2.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+print("\n--------------task 1.2---------------\n")
+""" тільки для дійсних значень і підмножин, без перехватів помилок неправильного вводу"""
+
```
Comment (kolyasalubov) [inline HW1/Olenka_M/HomeWork1.2.py:3]: good job

File: HW3/Olenka_M/HomeWork3.1.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import this, codecs
+get_zen = codecs.decode(this.s, 'rot13')
+
+#zen = ("%s" % get_zen)
+#zen = "{}".format(get_zen)
+zen = f"{get_zen}"
+
+print("----------------------------------------------\n")
+#print(zen)
+
```
Comment (kolyasalubov) [inline HW3/Olenka_M/HomeWork3.1.py:54]: good work

File: HW3/Olenka_M/HomeWork3.2.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+# product_of_digits = (int(digits_of_number[0]) * int(digits_of_number[1]) *
+# int(digits_of_number[2]) * int(digits_of_number[3]))
+# print(product_of_digits)
+
+# Version 0 end
+
+print("\n----------------Version 1----------------\n")
+
+# Version 1
+
```
Comment (kolyasalubov) [inline HW3/Olenka_M/HomeWork3.2.py:20]: good job

File: HW3/Olenka_M/HomeWork3.3.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+то можна послідувати попередньому способу"""
+
+""" Або якщо є можливість записати об'єкт в пам'яті без жодного вказівника на нього,
+то можна діяти таким способом"""
+
+""" Або якщо є можливість просто поміняти вказівники місцями, однією дією, 
+то теж хороший варіант. Бо якщо почерзі, то як я розумію, 
+після перенаправлення першого вказівника, об'єкт, на який він вказував перед цим 
+стане з лічильником нуль і зникне, тому вже буде неможливо на нього перенаправити другий вказівник"""
+print("a =", a)
```
Comment (kolyasalubov) [inline HW3/Olenka_M/HomeWork3.3.py:39]: a = 45
b = 100
a, b = b, a

print(f"a:  {a}, "b:  {b}")

File: HW4/Olenka_M/HomeWork4.1.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+entered_number = int(input("Enter not negative integer number for getting factorial of the number: "))
```
Comment (kolyasalubov) [inline HW4/Olenka_M/HomeWork4.1.py:1]: good job

File: HW4/Olenka_M/HomeWork4.2.py [hunk 7]
```diff
@@ -0,0 +1,7 @@
+integer_number_list = [1,2,6,7,9,45,-34,0]
+float_number_list = []
+
+for number in integer_number_list:
```
Comment (kolyasalubov) [inline HW4/Olenka_M/HomeWork4.2.py:4]: for number in integer_number_list:
       float_number_list.append(float(number))
print(float_number_list)

File: HW5/Olenka_M/HomeWork5.1.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
+# print("\n-------------------------------------Version 0-----------------------------------\n")
+
```
Comment (kolyasalubov) [inline HW5/Olenka_M/HomeWork5.1.py:2]: good job

File: HW5/Olenka_M/HomeWork5.1.py [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    elif number % 3 == 0:
+        divisible_by_3.append(number)
+    else:
+        not_divisible_by_2_and_3.append(number)
+
+print(f'Numbers, that are divisible by 2: {divisible_by_2}.\nNumbers, that are divisible by 3: {divisible_by_3}.\n\
+Rest of numbers are: {not_divisible_by_2_and_3}.')
+
+# print("\n------------------------------------Version 1------------------------------------\n")
+
```
Comment (kolyasalubov) [inline HW5/Olenka_M/HomeWork5.1.py:19]: good job

File: HW5/Olenka_M/HomeWork5.2.py [hunk 10]
```diff
@@ -0,0 +1,6 @@
+user_login = input("Please, enter your user_login: ")
```
Comment (kolyasalubov) [inline HW5/Olenka_M/HomeWork5.2.py:1]: good job

File: HW6/Olenka_M/HomeWork6.1.py [hunk 11]
```diff
@@ -0,0 +1,65 @@
+# What function of all this vertions we need to use depends on how and why we will use it
+# I didn`t write type of input data and output data in DocString, becouse I wrote it in first line of every function
+# print("\n-------------------------------------Version 0-----------------------------------\n")
+
```
Comment (kolyasalubov) [inline HW6/Olenka_M/HomeWork6.1.py:4]: good job

File: HW6/Olenka_M/HomeWork6.2.py [hunk 12]
```diff
@@ -0,0 +1,39 @@
+def rectangle_square(side_a: float, side_b: float) -> float:
```
Comment (kolyasalubov) [inline HW6/Olenka_M/HomeWork6.2.py:1]: good job

File: HW6/Olenka_M/HomeWork6.3.py [hunk 13]
```diff
@@ -0,0 +1,26 @@
+# def calculate_characters(some_string: str) -> dict:
+#     """ 
+#     Function calculates the number of characters included in a given string 
+#     and returns dict where keys are unique characters of given string
+#     and values are number of relevant character of given string
+#     """
+#     return {item: some_string.count(item) for item in some_string}
+
```
Comment (kolyasalubov) [inline HW6/Olenka_M/HomeWork6.3.py:8]: good job

File: HW6/Olenka_M/distance_between_points.py [hunk 14]
```diff
@@ -0,0 +1,3 @@
+def distance(x1: float, y1: float, x2: float, y2: float) -> float:
```
Comment (kolyasalubov) [inline HW6/Olenka_M/distance_between_points.py:1]: good job

File: HW6/Olenka_M/from_num_into_str.py [hunk 15]
```diff
@@ -0,0 +1,6 @@
+def number_to_string(num: float) -> str:
+    return str(num)
+
+def number_to_string(num: float) -> str:
```
Comment (kolyasalubov) [inline HW6/Olenka_M/from_num_into_str.py:4]: def number_to_string(num: float) -> str:
    return f"{str(num)}"

File: HW6/Olenka_M/infinite_loop.py [hunk 16]
```diff
@@ -0,0 +1,11 @@
+def create_array(n: float) -> list:
```
Comment (kolyasalubov) [inline HW6/Olenka_M/infinite_loop.py:1]: good work

File: HW6/Olenka_M/reverse_words.py [hunk 17]
```diff
@@ -0,0 +1,12 @@
+def reverse(st:str) -> str:
```
Comment (kolyasalubov) [inline HW6/Olenka_M/reverse_words.py:1]: good job

File: HW6/Olenka_M/secret_massage.py [hunk 18]
```diff
@@ -0,0 +1,8 @@
+def greet(name:str) -> str:
```
Comment (kolyasalubov) [inline HW6/Olenka_M/secret_massage.py:1]: good job

File: HW6/Olenka_M/string_transforming.py [hunk 19]
```diff
@@ -0,0 +1,3 @@
+def filter_words(st: str) -> str:
```
Comment (kolyasalubov) [inline HW6/Olenka_M/string_transforming.py:1]: good work

File: HW7/Olenka_M/HW_7_1_def_sqaure.py [hunk 20]
```diff
@@ -0,0 +1,21 @@
+def rectangle_square(side_a: float, side_b: float) -> float:
```
Comment (kolyasalubov) [inline HW7/Olenka_M/HW_7_1_def_sqaure.py:1]: good work

File: HW7/Olenka_M/HW_7_1_import.py [hunk 21]
```diff
@@ -0,0 +1,21 @@
+import HW_7_1_def_sqaure as sq
+
```
Comment (kolyasalubov) [inline HW7/Olenka_M/HW_7_1_import.py:2]: good job

File: HW7/Olenka_M/HW_7_2_validit_passward.py [hunk 22]
```diff
@@ -0,0 +1,30 @@
+def check_validity_password(password: str) -> str:
+    """
+    Function check whether password is relevant the next 
+    requirements:
+    At least 1 letter between [a-z] and 1 letter between [A-Z].
+    At least 1 number between [0-9].
+    At least 1 character from [$#@].
+    Minimum length 6 characters.
+    Maximum length 16 characters.
+    """
```
Comment (kolyasalubov) [inline HW7/Olenka_M/HW_7_2_validit_passward.py:10]: good job!

File: HW7/Olenka_M/boolean_result.py [hunk 23]
```diff
@@ -0,0 +1,6 @@
+def bool_to_word(boolean):
```
Comment (kolyasalubov) [inline HW7/Olenka_M/boolean_result.py:1]: good job

File: HW7/Olenka_M/count_and_sum_list_el-ts.py [hunk 24]
```diff
@@ -0,0 +1,13 @@
+def count_positives_sum_negatives(arr: list) -> list:
```
Comment (kolyasalubov) [inline HW7/Olenka_M/count_and_sum_list_el-ts.py:1]: good job

File: HW7/Olenka_M/count_true_in_array.py [hunk 25]
```diff
@@ -0,0 +1,2 @@
+def count_sheeps(sheep):
```
Comment (kolyasalubov) [inline HW7/Olenka_M/count_true_in_array.py:1]: good job

File: HW7/Olenka_M/divisible_3_or_5.py [hunk 26]
```diff
@@ -0,0 +1,11 @@
+def solution(number: int) -> int:
```
Comment (kolyasalubov) [inline HW7/Olenka_M/divisible_3_or_5.py:1]: good job

File: HW7/Olenka_M/first_letter_str.py [hunk 27]
```diff
@@ -0,0 +1,3 @@
+def are_you_playing_banjo(name):
```
Comment (kolyasalubov) [inline HW7/Olenka_M/first_letter_str.py:1]: good job

File: HW7/Olenka_M/fuel_left.py [hunk 28]
```diff
@@ -0,0 +1,6 @@
+def zero_fuel(distance_to_pump, mpg, fuel_left):
```
Comment (kolyasalubov) [inline HW7/Olenka_M/fuel_left.py:1]: good job

File: HW7/Olenka_M/last_letter.py [hunk 29]
```diff
@@ -0,0 +1,3 @@
+def correct_tail(body, tail):
```
Comment (kolyasalubov) [inline HW7/Olenka_M/last_letter.py:1]: good job

File: HW7/Olenka_M/reverse_list.py [hunk 30]
```diff
@@ -0,0 +1,6 @@
+def reverse_list(l):
```
Comment (kolyasalubov) [inline HW7/Olenka_M/reverse_list.py:1]: good job

File: HW3/Olenka_M/HomeWork3.3.py (hunk 5 above)
Comment (OlenkaMysko) [reply HW3/Olenka_M/HomeWork3.3.py:39]: 👍👍👍😍

**Pull request merged**